In [2]:
#instalando librerias necesarias 
!pip install pandas 

### Importación de datos



In [3]:
import pandas as pd

url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

tienda = pd.read_csv(url)
tienda2 = pd.read_csv(url2)
tienda3 = pd.read_csv(url3)
tienda4 = pd.read_csv(url4)

tienda.head()

Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      16/01/2021      Pedro Gomez          Bogotá             4   
1      18/05/2022  Beatriz Morales        Medellín             1   
2      15/03/2021   Juan Fernandez       Cartagena             1   
3      03/05/2022   Juan Fernandez            Cali             4   
4      07/11/2020    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon  
0  Tarjeta de crédito                   8   4.60971 -74.08175  
1  Tarjeta de crédito                   4   6.25184 -75.56359  
2  Tarjeta de crédito                   1  10.39972 -75.51444  
3               Nequi                   1   3.43722 -76.52250  
4               Nequi                   1   6.25184 -75.56359

# Explorando datos de cada tienda

In [4]:
#Tienda 1 
tienda.describe()

Precio  Costo de envío  Calificación  Cantidad de cuotas  \
count  2.359000e+03     2359.000000   2359.000000         2359.000000   
mean   4.878679e+05    26018.609580      3.976685            2.943196   
std    6.146868e+05    32860.001783      1.415370            2.819897   
min    7.600000e+03        0.000000      1.000000            1.000000   
25%    5.575000e+04     3100.000000      3.000000            1.000000   
50%    2.353000e+05    12400.000000      5.000000            1.000000   
75%    6.781000e+05    36000.000000      5.000000            4.000000   
max    2.977000e+06   160800.000000      5.000000           24.000000   

               lat          lon  
count  2359.000000  2359.000000  
mean      5.365283   -74.789417  
std       2.287445     1.217827  
min      -4.215280   -77.281110  
25%       4.609710   -75.563590  
50%       4.609710   -74.199040  
75%       6.251840   -74.081750  
max      11.544440   -67.923900

In [5]:
#Tienda 2
tienda2.describe()

Precio  Costo de envío  Calificación  Cantidad de cuotas  \
count  2.359000e+03     2359.000000   2359.000000         2359.000000   
mean   4.732274e+05    25216.235693      4.037304            2.902501   
std    5.955595e+05    31741.375166      1.356153            2.848416   
min    7.600000e+03        0.000000      1.000000            1.000000   
25%    5.455000e+04     2900.000000      4.000000            1.000000   
50%    2.182000e+05    11600.000000      5.000000            1.000000   
75%    6.777000e+05    35450.000000      5.000000            4.000000   
max    2.953400e+06   162500.000000      5.000000           24.000000   

               lat          lon  
count  2359.000000  2359.000000  
mean      5.499753   -74.742963  
std       2.351118     1.185997  
min      -4.215280   -77.281110  
25%       4.609710   -75.563590  
50%       4.609710   -74.199040  
75%       6.251840   -74.081750  
max      11.544440   -67.923900

In [6]:
#Tienda 3 
tienda3.describe()

Precio  Costo de envío  Calificación  Cantidad de cuotas  \
count  2.359000e+03     2359.000000   2359.000000         2359.000000   
mean   4.654598e+05    24805.680373      4.048326            2.967783   
std    5.806434e+05    30945.889626      1.386426            2.812153   
min    7.600000e+03        0.000000      1.000000            1.000000   
25%    5.740000e+04     3100.000000      4.000000            1.000000   
50%    2.263000e+05    11900.000000      5.000000            2.000000   
75%    6.660500e+05    34950.000000      5.000000            4.000000   
max    2.968200e+06   158400.000000      5.000000           24.000000   

               lat          lon  
count  2359.000000  2359.000000  
mean      5.437076   -74.733463  
std       2.405272     1.224572  
min      -4.215280   -77.281110  
25%       4.609710   -75.563590  
50%       4.609710   -74.199040  
75%       6.251840   -74.081750  
max      11.544440   -67.923900

In [7]:
#Tienda 4 
tienda4.describe()

Precio  Costo de envío  Calificación  Cantidad de cuotas  \
count  2.358000e+03     2358.000000   2358.000000         2358.000000   
mean   4.403629e+05    23459.457167      3.995759            2.896098   
std    5.687556e+05    30279.100836      1.403063            2.730393   
min    7.600000e+03        0.000000      1.000000            1.000000   
25%    5.340000e+04     2900.000000      3.000000            1.000000   
50%    2.094500e+05    10950.000000      5.000000            1.000000   
75%    6.342250e+05    33300.000000      5.000000            4.000000   
max    2.902200e+06   154700.000000      5.000000           18.000000   

               lat          lon  
count  2358.000000  2358.000000  
mean      5.413519   -74.774597  
std       2.376728     1.199749  
min      -4.215280   -77.281110  
25%       4.609710   -75.563590  
50%       4.609710   -74.199040  
75%       6.251840   -74.081750  
max      11.544440   -67.923900

## Tipos de datos 

In [8]:
# Información más detallada para cada tienda
#Tienda 1 : 
print("Información detallada - Tienda 1:")
tienda.info()

Información detallada - Tienda 1:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2359 entries, 0 to 2358
Data columns (total 12 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Producto                2359 non-null   object 
 1   Categoría del Producto  2359 non-null   object 
 2   Precio                  2359 non-null   float64
 3   Costo de envío          2359 non-null   float64
 4   Fecha de Compra         2359 non-null   object 
 5   Vendedor                2359 non-null   object 
 6   Lugar de Compra         2359 non-null   object 
 7   Calificación            2359 non-null   int64  
 8   Método de pago          2359 non-null   object 
 9   Cantidad de cuotas      2359 non-null   int64  
 10  lat                     2359 non-null   float64
 11  lon                     2359 non-null   float64
dtypes: float64(4), int64(2), object(6)
memory usage: 221.3+ KB


In [9]:
#Tienda 2 : 
print("\nInformación detallada - Tienda 2:")
tienda2.info()


Información detallada - Tienda 2:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2359 entries, 0 to 2358
Data columns (total 12 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Producto                2359 non-null   object 
 1   Categoría del Producto  2359 non-null   object 
 2   Precio                  2359 non-null   float64
 3   Costo de envío          2359 non-null   float64
 4   Fecha de Compra         2359 non-null   object 
 5   Vendedor                2359 non-null   object 
 6   Lugar de Compra         2359 non-null   object 
 7   Calificación            2359 non-null   int64  
 8   Método de pago          2359 non-null   object 
 9   Cantidad de cuotas      2359 non-null   int64  
 10  lat                     2359 non-null   float64
 11  lon                     2359 non-null   float64
dtypes: float64(4), int64(2), object(6)
memory usage: 221.3+ KB


In [10]:
#Tienda 3 : 
print("\nInformación detallada - Tienda 3:")
tienda3.info()


Información detallada - Tienda 3:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2359 entries, 0 to 2358
Data columns (total 12 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Producto                2359 non-null   object 
 1   Categoría del Producto  2359 non-null   object 
 2   Precio                  2359 non-null   float64
 3   Costo de envío          2359 non-null   float64
 4   Fecha de Compra         2359 non-null   object 
 5   Vendedor                2359 non-null   object 
 6   Lugar de Compra         2359 non-null   object 
 7   Calificación            2359 non-null   int64  
 8   Método de pago          2359 non-null   object 
 9   Cantidad de cuotas      2359 non-null   int64  
 10  lat                     2359 non-null   float64
 11  lon                     2359 non-null   float64
dtypes: float64(4), int64(2), object(6)
memory usage: 221.3+ KB


In [11]:
#Tienda 4: 
print("\nInformación detallada - Tienda 4:")
tienda4.info()


Información detallada - Tienda 4:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2358 entries, 0 to 2357
Data columns (total 12 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Producto                2358 non-null   object 
 1   Categoría del Producto  2358 non-null   object 
 2   Precio                  2358 non-null   float64
 3   Costo de envío          2358 non-null   float64
 4   Fecha de Compra         2358 non-null   object 
 5   Vendedor                2358 non-null   object 
 6   Lugar de Compra         2358 non-null   object 
 7   Calificación            2358 non-null   int64  
 8   Método de pago          2358 non-null   object 
 9   Cantidad de cuotas      2358 non-null   int64  
 10  lat                     2358 non-null   float64
 11  lon                     2358 non-null   float64
dtypes: float64(4), int64(2), object(6)
memory usage: 221.2+ KB


## Verificando si hay valores faltantes

Debido a que el RangeIndex es  `RangeIndex: 2358 entries, 0 to 2357` y `Non-Null Count` es `2358` significa que : 
> El DataFrame tiene 2358 filas en total  <br/>

> La columna específica tiene 2358 valores no nulos  <br/>

> Por lo tanto, no hay valores nulos o faltantes en esa columna  <br/>

# 1. Análisis de facturación



## Ingreso total por cada tienda

In [ ]:
facturacion_tienda=tienda['Precio'].sum()
facturacion_tienda

np.float64(1150880400.0)

In [13]:
facturacion_tienda2=tienda2['Precio'].sum()
facturacion_tienda2

np.float64(1116343500.0)

In [14]:
facturacion_tienda3=tienda3['Precio'].sum()
facturacion_tienda3

np.float64(1098019600.0)

In [15]:
facturacion_tienda4=tienda4['Precio'].sum()
facturacion_tienda4

np.float64(1038375700.0)

# 2. Ventas por categoría

In [ ]:
#Tienda 1: 
categorias_tienda=tienda['Categoría del Producto']
conteo_tienda={}
for item in categorias_tienda:
    if item in conteo_tienda: 
        conteo_tienda[item]+=1
    else:
        conteo_tienda[item]=1
print(conteo_tienda)

##Otro metodo funcional pero menos eficiente : 

"""
valores_unicos = set(tienda['Categoría del Producto'])
conteo_total = {valor: tienda['Categoría del Producto'].count(valor) for valor in valores_unicos}
print(conteo_total)

"""

{'Electrónicos': 448, 'Muebles': 465, 'Juguetes': 324, 'Electrodomésticos': 312, 'Artículos para el hogar': 171, 'Deportes y diversión': 284, 'Libros': 173, 'Instrumentos musicales': 182}


In [19]:
#Tienda 2: 
categorias_tienda2=tienda2['Categoría del Producto']
conteo_tienda2={}
for item in categorias_tienda2:
    if item in conteo_tienda2: 
        conteo_tienda2[item]+=1
    else:
        conteo_tienda2[item]=1
print(conteo_tienda2)


{'Muebles': 442, 'Instrumentos musicales': 224, 'Deportes y diversión': 275, 'Electrónicos': 422, 'Artículos para el hogar': 181, 'Juguetes': 313, 'Electrodomésticos': 305, 'Libros': 197}


In [20]:
#Tienda 3: 
categorias_tienda3=tienda3['Categoría del Producto']
conteo_tienda3={}
for item in categorias_tienda3:
    if item in conteo_tienda3: 
        conteo_tienda3[item]+=1
    else:
        conteo_tienda3[item]=1
print(conteo_tienda3)


{'Electrodomésticos': 278, 'Muebles': 499, 'Instrumentos musicales': 177, 'Deportes y diversión': 277, 'Electrónicos': 451, 'Juguetes': 315, 'Artículos para el hogar': 177, 'Libros': 185}


In [21]:
#Tienda 3: 
categorias_tienda4=tienda4['Categoría del Producto']
conteo_tienda4={}
for item in categorias_tienda4:
    if item in conteo_tienda4: 
        conteo_tienda4[item]+=1
    else:
        conteo_tienda4[item]=1
print(conteo_tienda4)

{'Electrodomésticos': 254, 'Muebles': 480, 'Deportes y diversión': 277, 'Libros': 187, 'Electrónicos': 451, 'Instrumentos musicales': 170, 'Juguetes': 338, 'Artículos para el hogar': 201}


# 3. Calificación promedio de la tienda


# 4. Productos más y menos vendidos

# 5. Envío promedio por tienda